src/components/AssetsPage/index.jsx

```diff
@@ -44,63 +37,41 @@ export class AssetsPage extends React.Component {
     }
   }
 
-  hasNoAssets = assetsList => (
-    assetsList.length === 0
-  )
+  getPageType = (props) => {
+    const numberOfAssets = props.assetsList.length;
 
-  hasAssets = assetsList => (
-    assetsList.length > 0
-  )
-
-  handleCheckBoxChange = (checked) => {
-    this.setState({
-      checked,
-    });
+    if ('type' in props.status && props.status.type === assetActions.request.REQUESTING_ASSETS) {
+      return this.state.pageType;
+    } else if (numberOfAssets > 0) {
+      return pageTypes.types.NORMAL;
+    } else if (numberOfAssets === 0 && hasSelectedFilters(props.filtersMetaData.assetTypes)) {
+      return pageTypes.types.NO_RESULTS;
+    }
+    return pageTypes.types.NO_ASSETS;
   }
 
-  renderAssetsDropZone = () => (
-    <WrappedAssetsDropZone
-      maxFileSizeMB={this.props.uploadSettings.max_file_size_in_mbs}
-    />
-  )
-
-  renderAssetsPage = () => (
-    <React.Fragment>
-      <div className={edxBootstrap.col}>
-        { this.renderAssetsDropZone() }
-        <WrappedAssetsFilters />
-      </div>
-      <div className={edxBootstrap['col-10']}>
-        <WrappedAssetsTable />
-        <WrappedPagination />
-      </div>
-    </React.Fragment>
-  )
-
-  renderNoAssetsPage = () => (
-    <React.Fragment>
-      <div className={edxBootstrap.col}>
-        { this.renderAssetsDropZone() }
-      </div>
-      <div className={edxBootstrap['col-10']}>
-        { !this.state.hasAssets && this.renderNoAssetsBody() }
-      </div>
-    </React.Fragment>
-  )
-
-  renderNoAssetsBody = () => (
-    <React.Fragment>
-      <h3>0 files in your course</h3>
-      <h4>Enhance your course content by uploading files such as images and documents.</h4>
-    </React.Fragment>
-  )
+  getPage = (type) => {
+    switch (type) {
+      case pageTypes.types.NORMAL:
+        return pageTypes.renderAssetsPage(this.props.uploadSettings.max_file_size_in_mbs);
+      case pageTypes.types.NO_ASSETS:
+        return pageTypes.renderNoAssetsPage(this.props.uploadSettings.max_file_size_in_mbs);
+      case pageTypes.types.NO_RESULTS:
+        return pageTypes.renderNoResultsPage(this.props.uploadSettings.max_file_size_in_mbs,
+          this.props.filtersMetaData, this.clearFilters);
+      case pageTypes.types.SKELETON:
+        return pageTypes.renderSkeletonPage(this.props.uploadSettings.max_file_size_in_mbs);
+      default:
+        return pageTypes.renderSkeletonPage(this.props.uploadSettings.max_file_size_in_mbs);
```

Comment:
The default is the SKELETON page. Does that seem reasonable? Or should it be the previous page type (à la line 43 of this file)?

Reply:
It doesn't seem like this component would know about the previous page type at this point. It only knows about the current `this.state.pageType` which is passed as `type` to this function. Since the constructor sets that to SKELETON, I don't think this default case should ever be hit.

Reply:
All right. Should the default case just return an empty Object? I like having the case for the SKELETON page be explicit, but I realize it could just be made the default case instead.

Reply:
Hmm an error would make the most sense here since it's a case we don't expect.

```
throw new Error(`Unknown pageType {type}`);
```